### PR TITLE
Switch to Versioneer.py for version string generation

### LINKED ISSUE #663: Kotekan version string inconsistent
For kotekan on a GPU node:
```
$ kotekan -v
Kotekan version HEAD.2019.12a.0-8000402a
```

On a recv node:
```
$ kotekan -v
Kotekan version HEAD.2019.12a.0-8000402
```
(note the lack of the trailing `a`)

### PR DESCRIPTION
**Pull request by james-s-willis**
Sets the `kotekan` version using `versioneer`. Closes #663.

### CODE REVIEW

**Comment by nritsche** [comment]
Does this obsolete the old version.py?

**Comment by james-s-willis** [comment]
> Does this obsolete the old version.py?

Ah good question. It does, I will remove the old `version.py`.

File: lib/version/MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,2 @@
+include versioneer.py
```
Comment (nritsche) [inline lib/version/MANIFEST.in]: I don't think we need this.

Comment (james-s-willis) [reply lib/version/MANIFEST.in]: The file? Or that line? I just followed the `versioneer` instructions and committed everything that was generated. Also, I didn't include a `setup.py` do you think I am okay without it? It seems fine without it.

Comment (nritsche) [reply lib/version/MANIFEST.in]: I mean the file. That's for distribution using pip etc.

Oh yeah, the version should also be set for the python kotekan package (in `python/setup.py`).

Comment (nritsche) [reply lib/version/MANIFEST.in]: Can you check that the rfi_receiver also ends up with the correct `__version__` from versioneer?

Comment (james-s-willis) [reply lib/version/MANIFEST.in]: > Can you check that the rfi_receiver also ends up with the correct `__version__` from versioneer?

```
[willis@scarlet rfi_receiver]$ python rfi_receiver.py
2019.12+26.gcfc745ee.dirty
```

Yeah that seems okay.

Comment (james-s-willis) [reply lib/version/MANIFEST.in]: > I mean the file. That's for distribution using pip etc.

Ah I see, I will delete that file then.
 
> Oh yeah, the version should also be set for the python kotekan package (in `python/setup.py`).

How do I pass this `setup.py` to `versioneer` from 'lib/version`?

Comment (nritsche) [reply lib/version/MANIFEST.in]: The `setup.py` is for installing the kotekan python package. At the moment it hardcodes a version number, so we should tell it where to get the versioneered version number. Actually, this should be done by versioneer automatically (https://github.com/warner/python-versioneer/blob/master/INSTALL.md#installation). Maybe it had troubles finding the `setup.py`.

Comment (andrerenard) [reply lib/version/MANIFEST.in]: Is this change going to require that we `pip install versioneer` before building kotekan?   If so we need to update the documentation.  We do already require `python-yaml` so we just need to make sure people install `versioneer` at the same time.

Comment (nritsche) [reply lib/version/MANIFEST.in]: The way it works is that versioneer itself is added to the kotekan repository (to prevent what you describe).

Comment (james-s-willis) [reply lib/version/MANIFEST.in]: > The `setup.py` is for installing the kotekan python package. At the moment it hardcodes a version number, so we should tell it where to get the versioneered version number. Actually, this should be done by versioneer automatically (https://github.com/warner/python-versioneer/blob/master/INSTALL.md#installation). Maybe it had troubles finding the `setup.py`.

```
def get_root():
    """Get the project root directory.

    We require that all commands are run from the project root, i.e. the
    directory that contains setup.py, setup.cfg, and versioneer.py .
    """
    root = os.path.realpath(os.path.abspath(os.getcwd()))
    setup_py = os.path.join(root, "setup.py")
    versioneer_py = os.path.join(root, "versioneer.py")
```

According to the documentation `versioneer` needs to be run in the same directory as `setup.py`, `setup.cfg` and `versioneer.py`. So I would need to move `versioneer.py` and `setup.cfg` to `python/`.

Comment (james-s-willis) [reply lib/version/MANIFEST.in]: My latest changes should fix this

File: python/print_kotekan_version.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Prints the kotekan version using versioneer
+# The script is called from make_version.cmake.in
+from _version import get_versions
+
+print get_versions()["version"]
```
Comment (nritsche) [inline python/print_kotekan_version.py]: This won't work in python 3
```
[  1%] Generating version.c

  File "/Users/jenkins/workspace/kotekan_PR-664/python/print_kotekan_version.py", line 5

    print get_versions()["version"]

                     ^

SyntaxError: invalid syntax
```
(https://lwlab.dunlap.utoronto.ca:18081/blue/organizations/jenkins/kotekan/detail/PR-664/13/pipeline/35)
```suggestion
print(get_versions()["version"])
```

**Comment by andrerenard** [comment]
I changed the name of the PR, but that seems to have triggered the CI to rebuild.  Will merge once that's done.

**Pull request merged**